gestalt/__init__.py

```diff
@@ -611,11 +596,60 @@ def fetch_vault_secrets(self) -> None:
             try:
                 secret_token = self.vault_client.secrets.kv.v2.read_secret_version(
                     mount_point=str(mount_path), path=secret_path)
+                print(secret_token)
                 self.__conf_data.update(secret_token['data']['data'])
+                if secret_token['lease_id'] != '': 
+                    secret_lease = (secret_token['lease_id'],
+                                secret_token['lease_duration'], time.time())
+                    self.secret_ttl_identifier.append(secret_lease)
+                
             except hvac.exceptions.InvalidPath as err:
                 raise RuntimeError(
                     "Gestalt Error: The secret path or mount is set incorrectly"
                 )
             except requests.exceptions.ConnectionError as err:
                 raise RuntimeError(
                     "Gestalt Error: Gestalt couldn't connect to Vault")
+            except Exception as err:
+                raise RuntimeError(f"Gestalt Error: {err}")
+
+    def generate_database_dynamic_secret(self, mount_point: str, db_name: str, role_name: str):
```

Comment:
This should't be so specific. All secrets backends use the same API, so there is no reason to treat this any differently if it's AWS, Postgres, etc. 

In addition, KV1, Dynamic Secrets Engines, and KV2 are actually the same response schema, but kv2 of course nests metadata and an additional data in the response.data. I understand there are opinionated methods in hvac, but you should actually be able to use KV1 fetch for ALL vault paths, then inspect it to figure out your next steps. Mentioning all of this so that you have a clearer picture as you implement functionality.